sky/data/storage.py

```diff
@@ -822,13 +858,15 @@ def upload(self):
             StorageUploadError: if upload fails.
         """
         try:
-            if self.source is not None:
+            if isinstance(self.source, list):
```

Comment:
Can we abstract this to `AbstractStore`  : `batch_sync(...)`

Reply:
Moved it to data_utils.py!